app.js

```diff
@@ -1,101 +1,94 @@
+
+const parseString = require('xml2js').parseString;
+const requestP    = require('request-promise-native');
+const CronJob     = require('cron').CronJob;
+const NodeID3     = require('node-id3');
+const concat      = require('concat-stream');
+const rimraf      = require('rimraf');
+const axios       = require('axios');
+const fs          = require('fs');
+
+const { 
+    log,
+    sanitizeContent,
+    sanitizeEpisode
+}                  = require('./lib/helpers');
+const TwController = require('./controllers/twitter.controller');
+const DbController = require('./controllers/db.controller');
+
+const TEST_CHANNEL = process.env.TEST_CHANNEL;
 
 const COVER = './assets/cover.jpg'
 const DDIR  = './downloads/'
-
-new CronJob('0 */30 * * * *', () => {
-    main()
-}, null, true)
+const CRON  = process.env.CRON;
+const ENV   = process.env.NODE_ENV;
+const DB    = new DbController();
 
 /**
  * Main Application logic
  */
-function main() {
-    cleanDownloads()
-    .then(() => {
-       return getRssList()
-    }).then(res => {
-        if (res.length) {
-            return getStoredPodcasts()
-            .then(storedPodcasts => {
-
-                eachOf(res, (value, key, callback) => {
-                    let { url, channel } = value
-
-                    getFeed(url)
-                    .then(feed => {
-                        return ignoreUploadedPodcasts(feed, storedPodcasts)
-                    }).then(feed => {
-                        return parseFeed(feed)
-                    }).then(feed => {
-                        return sendFeedToTelegram(feed, channel)
-                    }).then(() => {
-                        callback()
-                    }).catch((error) => {
-                        callback(error)
-                    })
-
-                }, err => {
-                    if (err) logger(false, `Some generic error situation going on here: ${err}`)
-                })
-            })
+async function main() {
+    try {
+        await cleanDownloads();
+    } catch (error) {
+        log('Error while cleaning Downloads folder.', error);
+    }
+
+    try {
+        const rssList = await DB.getRssList();
+    
+        if (rssList.length) {
+            const storedPodcasts = await DB.getStoredPodcasts();
+    
+            while (!!rssList.length) {
+                const rssUrl = rssList.pop();
+                const { url, channel } = rssUrl
+    
+                try {
+                    let feed = await getFeed(url);
+                    feed = ignoreUploadedPodcasts(feed, storedPodcasts);
+        
+                    const parsedFeed = parseFeed(feed)
+        
+                    if (!!parsedFeed.episodes.length) {
+                        await sendFeedToTelegram(parsedFeed, channel)
+                    } else {
+                        log(`${channel} parseFeed`, 'Nothing to upload');
+                    }
+                } catch (error) {
+                    log('Error getting feeds:', error);
```

Comment:
Using the one `try/catch` approach, you can throw the errors in every method and only display it or take care of them in one place.